src/path.jl

```diff
@@ -687,8 +687,8 @@ Download a file from the remote `url` and save it to the `localfile` path.
 NOTE: Not downloading into a `localfile` directory matches the base Julia behaviour.
 https://github.com/rofinn/FilePathsBase.jl/issues/48
 """
-function Base.download(url::AbstractString, localfile::P) where P <: AbstractPath
-    mktemp(P) do fp, io
+function Base.download(url::AbstractString, localfile::AbstractPath)
+    mktmp() do fp, io
```

Comment:
`mktemp(P)` previously returned a `SystemPath` by default. We've changed this, so we've made this explicit to avoid infinite recursion on the `download` call below.